Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2005-01-04
Initial Package Version: 20050103
Origin: Linux-MIPS Mailing List
Upstream Status: Unknown
Description: Fixes Compile Issues with Current GLIBC
 
diff -Naur glibc-20050103.orig/csu/Makefile glibc-20050103/csu/Makefile
--- glibc-20050103.orig/csu/Makefile	2004-08-20 20:12:43.000000000 +0000
+++ glibc-20050103/csu/Makefile	2005-01-05 07:29:47.200094488 +0000
@@ -146,7 +146,7 @@
 subdir_lib: $(extra-objs:%=$(objpfx)%)
 
 define link-relocatable
-$(CC) -nostdlib -nostartfiles -r -o $@ $^
+$(CC) $(ABI_FLAG) -nostdlib -nostartfiles -r -o $@ $^
 endef
 
 ifndef start-installed-name-rule
diff -Naur glibc-20050103.orig/sysdeps/mips/fpu_control.h glibc-20050103/sysdeps/mips/fpu_control.h
--- glibc-20050103.orig/sysdeps/mips/fpu_control.h	2001-07-06 04:56:00.000000000 +0000
+++ glibc-20050103/sysdeps/mips/fpu_control.h	2005-01-05 07:29:47.202094184 +0000
@@ -74,7 +74,7 @@
 #define _FPU_RC_UP      0x2
 #define _FPU_RC_DOWN    0x3
 
-#define _FPU_RESERVED 0xfe3c0000  /* Reserved bits in cw */
+#define _FPU_RESERVED   0xfebc0000  /* Reserved bits in cw */
 
 
 /* The fdlibm code requires strict IEEE double precision arithmetic,
diff -Naur glibc-20050103.orig/sysdeps/mips/mips64/n32/Makefile glibc-20050103/sysdeps/mips/mips64/n32/Makefile
--- glibc-20050103.orig/sysdeps/mips/mips64/n32/Makefile	2003-03-29 08:15:28.000000000 +0000
+++ glibc-20050103/sysdeps/mips/mips64/n32/Makefile	2005-01-05 07:29:47.204093880 +0000
@@ -4,3 +4,7 @@
 ifeq ($(filter -mabi=n32,$(CC)),)
 CC += -mabi=n32
 endif
+ifeq ($(filter "-Wl,-m,elf32btsmipn32",$(LD)),)
+LD += -Wl,-m,elf32btsmipn32
+endif
+ABI_FLAG= -Wl,-m,elf32btsmipn32
diff -Naur glibc-20050103.orig/sysdeps/mips/mips64/n64/Makefile glibc-20050103/sysdeps/mips/mips64/n64/Makefile
--- glibc-20050103.orig/sysdeps/mips/mips64/n64/Makefile	2003-03-29 08:15:28.000000000 +0000
+++ glibc-20050103/sysdeps/mips/mips64/n64/Makefile	2005-01-05 07:29:47.205093728 +0000
@@ -4,3 +4,10 @@
 ifeq ($(filter -mabi=64,$(CC)),)
 CC += -mabi=64
 endif
+ifeq ($(filter "-Wl,-m,elf64btsmip",$(LDFLAGS)),)
+LDFLAGS += -Wl,-m,elf64btsmip
+endif
+ifeq ($(filter "-Wl,-m,elf64btsmip",$(ASFLAGS-.os)),)
+ASFLAGS-.os += -Wl,-m,elf64btsmip
+endif
+ABI_FLAG= -Wl,-m,elf64btsmip
diff -Naur glibc-20050103.orig/sysdeps/mips/sys/ucontext.h glibc-20050103/sysdeps/mips/sys/ucontext.h
--- glibc-20050103.orig/sysdeps/mips/sys/ucontext.h	2004-11-24 04:36:10.000000000 +0000
+++ glibc-20050103/sysdeps/mips/sys/ucontext.h	2005-01-05 07:29:47.208093272 +0000
@@ -24,6 +24,7 @@
 #include <features.h>
 #include <sgidefs.h>
 #include <signal.h>
+#include <sgidefs.h>
 
 /* Type for general register.  */
 #if _MIPS_SIM == _ABIO32
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/Versions glibc-20050103/sysdeps/unix/sysv/linux/mips/Versions
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/Versions	2002-11-06 02:53:24.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/Versions	2005-01-05 07:29:47.210092968 +0000
@@ -29,3 +29,9 @@
     _test_and_set;
   }
 }
+librt {
+  GLIBC_2.0 {
+    # c*
+    clock_gettime; clock_settime;
+  }
+}
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips32/kern64/sysdep.h glibc-20050103/sysdeps/unix/sysv/linux/mips/mips32/kern64/sysdep.h
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips32/kern64/sysdep.h	2003-03-29 08:15:29.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/mips32/kern64/sysdep.h	2005-01-05 07:29:47.212092664 +0000
@@ -28,9 +28,9 @@
    so we have to redefine the `SYS_ify' macro here.  */
 #undef SYS_ify
 #ifdef __STDC__
-# define SYS_ify(syscall_name)	__NR_O32_##syscall_name
+# define SYS_ify(syscall_name)	__NR_##syscall_name
 #else
-# define SYS_ify(syscall_name)	__NR_O32_/**/syscall_name
+# define SYS_ify(syscall_name)	__NR_/**/syscall_name
 #endif
 
 #endif /* linux/mips/mips32/kern64/sysdep.h */
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/n32/sysdep.h glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/n32/sysdep.h
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/n32/sysdep.h	2004-10-18 05:16:07.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/n32/sysdep.h	2005-01-05 07:29:47.215092208 +0000
@@ -28,9 +28,9 @@
    so we have to redefine the `SYS_ify' macro here.  */
 #undef SYS_ify
 #ifdef __STDC__
-# define SYS_ify(syscall_name)	__NR_N32_##syscall_name
+# define SYS_ify(syscall_name)	__NR_##syscall_name
 #else
-# define SYS_ify(syscall_name)	__NR_N32_/**/syscall_name
+# define SYS_ify(syscall_name)	__NR_/**/syscall_name
 #endif
 
 #ifdef __ASSEMBLER__
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/n64/sysdep.h glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/n64/sysdep.h
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/n64/sysdep.h	2004-10-18 05:16:08.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/n64/sysdep.h	2005-01-05 07:29:47.217091904 +0000
@@ -28,9 +28,9 @@
    so we have to redefine the `SYS_ify' macro here.  */
 #undef SYS_ify
 #ifdef __STDC__
-# define SYS_ify(syscall_name)	__NR_N64_##syscall_name
+# define SYS_ify(syscall_name)	__NR_##syscall_name
 #else
-# define SYS_ify(syscall_name)	__NR_N64_/**/syscall_name
+# define SYS_ify(syscall_name)	__NR_/**/syscall_name
 #endif
 
 #ifdef __ASSEMBLER__
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/syscalls.list glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/syscalls.list
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/mips64/syscalls.list	2004-07-21 06:13:15.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/mips64/syscalls.list	2005-01-05 07:29:47.219091600 +0000
@@ -19,3 +19,4 @@
 semtimedop	-	semtimedop	i:ipip	semtimedop
 semget		-	semget		i:iii	__semget	semget
 semctl		-	semctl		i:iiii	__semctl	semctl
+semtimedop	-	semtimedop	i:ipip	semtimedop
diff -Naur glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/sys/syscall.h glibc-20050103/sysdeps/unix/sysv/linux/mips/sys/syscall.h
--- glibc-20050103.orig/sysdeps/unix/sysv/linux/mips/sys/syscall.h	2003-03-17 15:50:05.000000000 +0000
+++ glibc-20050103/sysdeps/unix/sysv/linux/mips/sys/syscall.h	2005-01-05 07:29:47.221091296 +0000
@@ -19,17 +19,10 @@
 #ifndef _SYSCALL_H
 #define _SYSCALL_H	1
 
-/* This file should list the numbers of the system the system knows.
-   But instead of duplicating this we use the information available
-   from the kernel sources.  */
-#ifdef _LIBC
-/* Since the kernel doesn't define macro names in a way usable for
-   glibc, we preprocess this header, and use it during the glibc build
-   process.  */
-# include <asm-unistd.h>
-#else
+/* This file should list the numbers of the system calls the system
+   knows. But instead of duplicating this we use the information
+   available from the kernel sources.  */
 # include <asm/unistd.h>
-#endif
 
 #ifndef _LIBC
 /* The Linux kernel header file defines macros `__NR_<name>', but some
